tests/test-timing.js

```diff
@@ -0,0 +1,62 @@
+'use strict'
+
+var http = require('http')
+  , server = require('./server')
+  , request = require('../index')
+  , tape = require('tape')
+
+var plain_server = server.createServer()
+  , redirect_mock_time = 10
+
+tape('setup', function(t) {
+  plain_server.listen(plain_server.port, function() {
+    plain_server.on('/', function (req, res) {
+      res.writeHead(200)
+      res.end('plain')
+    })
+    plain_server.on('/redir', function (req, res) {
+      // fake redirect delay to ensure strong signal for rollup check
+      setTimeout(function() {
+        res.writeHead(301, { 'location': 'http://localhost:' + plain_server.port + '/' })
+        res.end()
+      }, redirect_mock_time)
+    })
+
+    request('http://localhost:' + plain_server.port + '/', {}, function(err, res, body) {
+      t.equal(err, null)
```

Comment:
to ensure that, when redirects are handled inline, the time reported on the response is the sum of all request times, rather than just the time taken by the last request/response cycle